org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/LambdaExpression.java

```diff
@@ -560,9 +560,23 @@ private void analyzeExceptions() {
 		boolean oldAnalyseResources = compilerOptions.analyseResourceLeaks;
 		compilerOptions.analyseResourceLeaks = false;
 		try {
+			int firstLocalLocal = this.scope.outerMostMethodScope().analysisIndex;
+			LocalVariableBinding[] locals = this.scope.locals;
+			if (locals != null) {
```

Comment:
I was going to flag this null check as unnecessary, until I found that CodeSnippetScope and ModuleScope could indeed be constructed without assigning `locals`.
That in itself looks a bit like a bad smell (but surely modules have no locals ...). Anyway, this doesn't affect the current PR. Just mentioning.